public/timeout.js

```diff
@@ -0,0 +1,161 @@
+  sessionLifetime = (totalTime / 1000); //in seconds
+}
+
+function initTimeoutIdentifier() {
+   //set unique timeout countdown tracking interval id
+   timeoutGUID = (tokenInfo && tokenInfo.jti ? tokenInfo.jti : _createUUID());
+}
+
+/*
+ * unique timeout interval identifier
+ */
+function getTimeoutIdentifier() {
+  return "timeOnLoad_"+timeoutGUID;
+}
+
+/*
+ * reset timeout countdown
+ */
+function resetTimeout() {
+  clearInterval(timeoutIntervalId);
+  window.localStorage.removeItem(getTimeoutIdentifier());
+}
+/*
+ * set initial app access time when app loads
+ */
+function setTimeOnLoad() {
+  window.localStorage.setItem(getTimeoutIdentifier(), Date.now());
+}
+/*
+ * return the initial app access time
+ */
+function getLastActiveTime(){
+  let storedActiveTime = window.localStorage.getItem(getTimeoutIdentifier());
+  return storedActiveTime ? storedActiveTime : Date.now();
+}
+/*
+* check how much time has elapsed, if exceeds session lifetime, redirect to specified logout location
+*/
+function checkTimeout() {
+  let timeElapsed = (Date.now() - getLastActiveTime()) / 1000;
+  console.log("time elapsed since first visiting ", timeElapsed);
+  if (timeElapsed > sessionLifetime) { //session has expired
+    //logout user?
+    window.location = logoutLocation;
+  }
+}
+
+function isDOMReady() {
+  var targetNode = document.querySelector(".landing");
+  if (!targetNode) return false;
+  clearInterval(waitForDOMIntervalId);
+  return true;
+}
+
+/*
+ * initialized timeout interval countdown timer
+ */
+function startTimeoutTimer() {
+  var domElement = document.querySelector(".landing");
+  var customLogoutLocationAttr = domElement ? domElement.getAttribute("logoutlocation") : null;
+  if (customLogoutLocationAttr) logoutLocation = customLogoutLocationAttr;
+  clearTimeout(timeoutIntervalId);
+  console.log("timeout counting starts ");
+  //get expiration date/time to determine how long a session is?
+  getSessionTokenInfo();
+  //get the lifetime of a session
+  setSessionMaxTime();
+  //record timestamp that the site is first accessed
+  setTimeOnLoad();
+  //check whether session has expired every 15 seconds
+  timeoutIntervalId = setInterval(checkTimeout, 15000);
+}
+
+function _createUUID() {
+  return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+     var r = Math.random() * 16 | 0, v = c == 'x' ? r : (r & 0x3 | 0x8);
+     return v.toString(16);
+  });
+}
+
+/*
+ * initialize user events that will reset the timeout timer
+ * will not logout user when is active
+ */
+//TODO figure out what to do here, if user is active but access token expires?
+// currently just reset countdown without logging out user
+function resetTimeoutEvents() {
+  document.querySelector(".App").addEventListener("click", function() {
+    startTimeoutTimer();
+  });
+  document.addEventListener("scroll", function() {
+    window.requestAnimationFrame(function() {
+      console.log("time out interval reset");
```

Comment:
This is very helpful for debugging, but probably noisy otherwise. Would it be possible to only enable when SYSTEM_TYPE is not `production`?